src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Microsoft.Android.Sdk.Bindings.Gradle.targets

```diff
@@ -0,0 +1,102 @@
+
+This file contains MSBuild targets that support building and operating on Android Gradle projects.
+
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="Xamarin.Android.Tasks.Gradle"              AssemblyFile="Xamarin.Android.Build.Tasks.dll" />
+
+  <PropertyGroup>
+    <_BuildAndroidGradleProjectsStamp>$(_AndroidStampDirectory)_BuildAndroidGradleProjects.stamp</_BuildAndroidGradleProjectsStamp>
+    <_AGPOutputCacheFile>agpoutputs.cache</_AGPOutputCacheFile>
+  </PropertyGroup>
+
+  <ItemDefinitionGroup>
+    <AndroidGradleProjectReference>
+      <Bind>true</Bind>
+      <Configuration>Release</Configuration>
+      <ModuleName></ModuleName>
+      <OutputPath>$(MSBuildProjectDirectory)/$(IntermediateOutputPath)gradle/%(Filename)/</OutputPath>
+      <Pack>true</Pack>
+      <ReferenceLibraryOutputs>true</ReferenceLibraryOutputs>
+      <Visible>false</Visible>
+    </AndroidGradleProjectReference>
+  </ItemDefinitionGroup>
+
+  <Target Name="_GetBuildAndroidGradleProjectsInputs"
+        Condition=" '@(AndroidGradleProjectReference->Count())' != '0' ">
+    <ItemGroup>
+      <_AllGradleFiles Include="$([System.IO.Directory]::GetFiles('%(AndroidGradleProjectReference.FullPath)', '*.*', System.IO.SearchOption.AllDirectories))"
+          Condition= " Exists('%(AndroidGradleProjectReference.FullPath)') "/>
+      <_GradleInputs Include="@(_AllGradleFiles)"
+          Condition="  '%(Extension)' == '.java'
+                    or '%(Extension)' == '.kt'
+                    or '%(Extension)' == '.kts'
+                    or '%(Extension)' == '.gradle'
+                    or '%(Extension)' == '.properties'
+                    or '%(Extension)' == '.xml' " />
+      <_GradleInputs Include="$(MSBuildProjectFullPath)" />
+      <_GradleInputs Include="@(_AndroidMSBuildAllProjects)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="BuildAndroidGradleProjects"
+      Condition=" '@(AndroidGradleProjectReference->Count())' != '0' "
+      DependsOnTargets="_SetupMSBuildAllProjects;AndroidPrepareForBuild;_GetBuildAndroidGradleProjectsInputs"
+      Inputs="@(_GradleInputs)"
+      Outputs="$(_BuildAndroidGradleProjectsStamp)" >
+    <!-- Delete outputs folder if inputs have changed to force a partial rebuild -->
+    <RemoveDir Directories="@(AndroidGradleProjectReference->'%(OutputPath)outputs')" />
+
+    <!-- Run assemble task for project outputs, android app and library project outputs are currently supported -->
+    <Gradle ToolPath="%(AndroidGradleProjectReference.FullPath)"
+        Command="assemble%(AndroidGradleProjectReference.Configuration)"
+        ModuleName="%(AndroidGradleProjectReference.ModuleName)"
+        OutputPath="%(AndroidGradleProjectReference.OutputPath)"
+        IntermediateOutputPath="$(MSBuildProjectDirectory)/$(IntermediateOutputPath)"
```

Comment:
This feels "odd", in that we don't use this construct `$(MSBuildProjectDirectory)/$(IntermediateOutputPath)` anywhere else.  Could this instead "just" be `$(IntermediateOutputPath)`?

Reply:
I've replaced this with a parameter that specifies the full path to the init script being used to control the projects output directory.